tests/protocol/mockwiredata.py

```diff
@@ -182,3 +200,52 @@ def mock_gen_trans_cert(self, trans_prv_file, trans_cert_file):
 
         with open(trans_cert_file, 'w+') as cert_file:
             cert_file.write(self.trans_cert)
+
+    #
+    # Having trouble reading the regular expressions below? you are not alone!
```

Comment:
Thanks for adding this small guide on regular expressions and the helper functions, they'll be really helpful! :)

Reply:
Today, tests do simple text substitution, without adding any context for the substitution. I decided to use regexes to provide some context for the text replacement.

The new setters I created provide an abstraction over these mechanics and the implementation could be changed in the future if we find that we need full xml parsing via one of the standard libraries.